### DISCUSSION TITLE
How do we just use Fooocus v2 as addition and not the UI

### DISCUSSION THREAD

**Discussion by sdave-connexion** [lllyasviel/Fooocus#2525] [Q&A]
This UI is too nice but seems like I was a little late to find this one. I am working on a project and now I can't change things in the process. 

what I need is just the extension of Fooocus v2  in my current setup, I mean when the user writes the caption in my gradio app, I want it to trigger a function that uses Fooocus and enhances my input text basically.

is it possible to do it? 

```
import gradio as gr

def generate_description(text):
    return text

def generate_image(text):
    # Replace 'placeholder_image_path.png' with the path to your actual placeholder image
    return 'placeholder_image.png'

with gr.Blocks() as demo:
    with gr.Row():
        text_input = gr.Textbox(label="Input Text")
        description_output = gr.Textbox(label="Generated Description")
        image_output = gr.Image(label="Generated Image")
    
    generate_description_btn = gr.Button("Generate Description")
    generate_image_btn = gr.Button("Generate Image")

    generate_description_btn.click(generate_description, inputs=text_input, outputs=description_output)
    generate_image_btn.click(generate_image, inputs=text_input, outputs=image_output)

if __name__ == "__main__":
    demo.launch(share=True)

```

Basically, updating generate description function with a Fooocus-enriched prompt. 

Thank you in advance for any suggestion

Shantanu

**Comment by mashb1t** [comment]
sure, you can c&p or import the prompt extension logic as well as BLIP (image description).

**Reply by sdave-connexion** [reply]
Would be lovely if you could guide me more on how to integrate that with my gradio app, there are tons of .py files in Fooocus and many are interlinked so I am not sure how to do it. Any example article would be the best. Thanks

**Reply by mashb1t** [reply]
please find the prompt expansion in https://github.com/lllyasviel/Fooocus/blob/main/extras/expansion.py and the implementation in https://github.com/lllyasviel/Fooocus/blob/main/modules/async_worker.py#L473 + https://github.com/lllyasviel/Fooocus/blob/main/modules/default_pipeline.py#L247

You now should only have to copy the code and model over to your project!
